src/halmos/traces.py

```diff
@@ -0,0 +1,160 @@
+import io
```

Comment:
no changes here, just moving trace-related functionality out of `__main__.py` and into its own module